app/settings/data-export/data-export.controller.js

```diff
@@ -0,0 +1,103 @@
+module.exports = [
+    '$scope',
+    '$rootScope',
+    '$location',
+    'DataExport',
+    'Notify',
+    'FormEndpoint',
+    'FormAttributeEndpoint',
+    '$q',
+    '_',
+    'LoadingProgress',
+function (
+    $scope,
+    $rootScope,
+    $location,
+    DataExport,
+    Notify,
+    FormEndpoint,
+    FormAttributeEndpoint,
+    $q,
+    _,
+    LoadingProgress
+) {
+    $scope.exportAll = exportAll;
+    $scope.showFields = false;
+    $scope.showProgress = false;
+    $scope.selectFields = selectFields;
+    $scope.selectedFields = [];
+    $scope.exportSelected = exportSelected;
+    $scope.selectAll = selectAll;
+    $scope.displayedForms = [];
+    $scope.getForms = getForms;
+    $scope.isLoading = LoadingProgress.getLoadingState;
+    $scope.attachAttributes = attachAttributes;
+
+    // hiding export-in-progress screen when export is complete
+    $rootScope.$on('event:data_export:complete', function () {
+        $scope.showProgress = false;
+    });
+
+    // Redirect to home if not authorized
+    if ($rootScope.hasPermission('Bulk Data Import') === false) {
+        return $location.path('/');
+    }
+    // Change layout class
+    $rootScope.setLayout('layout-c');
+    // Change mode
+    $scope.$emit('event:mode:change', 'settings');
+
+    function exportAll() {
+        DataExport.prepareExport({format: 'csv'});
+        $scope.showProgress = true;
+    }
+
+    function selectFields() {
+        $scope.showFields = !$scope.showFields;
+    }
+
+    function getForms() {
+        FormEndpoint.queryFresh().$promise.then(function (response) {
+            $scope.forms = response;
+            $scope.attachAttributes();
+        });
+    }
+    function attachAttributes() {
+        var attributeQueries = [];
+        _.each($scope.forms, function (form) {
+            attributeQueries.push(FormAttributeEndpoint.query({formId: form.id}));
+        });
+        $q.all(attributeQueries).then(function (results) {
+            _.each(results, function (result, index) {
+                /* TODO: I am not sure if we can do this, can I trust the results come in the same order
+                so I can use the result index with the $scope.form[index]? Or do
+                we need to go through form-stage-id? :scream: */
+                $scope.forms[index].attributes = result;
```

Comment:
I would use the formstage id, there is no guarantee the forms are ordered by result relative to form, you will need to check the form id to be certain

Reply:
I suspected that but was hoping for the best :D I'll see if I find a better solution...

Reply:
Fixed it!